### PR TITLE
[WIP] Script the deployment of Code Docs

### PR DESCRIPTION
**Pull request by jamesbvaughan**
* **What does this PR do?**
  - adds a new step in the script in `.travis.yml` to deploy updated code docs

* **Are there points in the code the reviewer needs to double check?**
  - all of `.travis/push-docs.sh`
  - my changes in `.travis.yml`

* **What are the relevant issue numbers?**
  - #1152

### CODE REVIEW

**Comment by jamesbvaughan** [comment]
@flatcap I just finished midterm exams and had some time this morning to work on this. Let me know if I'm headed in the right direction and going about this the right way.

File: .travis.yml [hunk 1]
```diff
@@ -29,4 +29,5 @@ script:
   - doxygen doxygen/doxygen.conf
   - grep -v "$IGNORE" doxygen-build.txt | tee tmp.txt
   - test ! -s tmp.txt
+  - .travis/push-docs.sh
```
Comment (toogley) [inline .travis.yml]: As far as i understand, this is only generating the doxygen related things, right? I think the name `push-docs.sh` is misleading then, as it doesn't touch our [our neomutt guide](https://neomutt.org/guide/index.html).

File: .travis.yml [hunk 2]
```diff
@@ -29,4 +29,5 @@ script:
   - doxygen doxygen/doxygen.conf
```
Comment (toogley) [inline .travis.yml]: can we replace `addrbook.c body.c complete.c compress.c [...]`  with `./*.c` ? 

Might have nothing to do with your PR, but maybe we could fix that here.

File: .travis/push-docs.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# exit early if any command fails
+set -e
+
+git clone --recursive https://github.com/neomutt/neomutt.github.io
+
+(
+cd neomutt.github.io/code
+
+# Remove the old docs
+git rm -r ./*
```
Comment (toogley) [inline .travis/push-docs.sh]: As that makes the travis log currently hard to read, you might want to silence the output of this line.

File: .travis.yml (hunk 2 above)
Comment (flatcap) [reply .travis.yml]: This list will become `*.c` in the future.
For now, it's the files that I've tidied.

**Comment by flatcap** [comment]
I've just created two new repos:
- https://github.com/neomutt/test-neomutt.github.io
  An exact copy of the website
- https://github.com/neomutt/test-doxygen
  Empty repo (the original's a bit big for playing around with)

The two repos have deployment keys on them
- travis-deploy-doxygen to push to test-doxygen
- travis-deploy-website to push to test-neomutt.github.io

I've put the encrypted keys in `.travis/`

Using these repos and keys you should be able to push.

**Comment by flatcap** [comment]
Hi @jamesbvaughan 
Thanks for your help on this.
I finished off the last few details, and now it's up and ~running~ automating :-)

**Pull request closed without merging**